### ISSUE TITLE
Compile issue on Ubuntu 18.04 with Anaconda Python

### ISSUE THREAD

**Issue by alanwillwong** [ornladios/ADIOS2#2872]
**Description:**
I tried to compile adios2.7.1 with Python option turned on, but encountered an issue finding the Python/Numpy location.

**System:**
I'm using Ubuntu 18.04 and I have Python installed with anaconda3, at /opt/anaconda3/

I tried to compile adios2 with the command:

$>export CMAKE_PREFIX_PATH=/opt/anaconda3/lib/python3.7/site-packages
$>cmake -DADIOS2_USE_Python=ON ../

**Error Message:**
I received an error message not finding python or numpy, but both numpy and mpi4py works fine with the found Python 3.7.7. I also tried to export CMAKE_PREFIX_PATH, but still I receive the same error message.
![1631908137(1)](https://user-images.githubusercontent.com/59586135/133845487-c354fefb-cc7b-434c-a405-88fea3319feb.png)

**Comment by williamfgc** [comment]
Try removing `$>export CMAKE_PREFIX_PATH=/opt/anaconda3/lib/python3.7/site-packages` and use the solution reported [here](https://stackoverflow.com/a/36439709/470441) for anaconda (Python 2 is not supported). Make sure numpy is installed. Hope it helps.

**Comment by williamfgc** [comment]
FYI, adios2 has a conda distribution https://anaconda.org/conda-forge/adios2

**Comment by alanwillwong** [comment]
> FYI, adios2 has a conda distribution https://anaconda.org/conda-forge/adios2
Thank you for your information, I have tried this before and it worked fine. I want to install a newer standalone version, so we can have a well-organized system environment. Later we may need adios2 path for other programs as well, such as effis.

**Comment by alanwillwong** [comment]
> Try removing `$>export CMAKE_PREFIX_PATH=/opt/anaconda3/lib/python3.7/site-packages` and use the solution reported [here](https://stackoverflow.com/a/36439709/470441) for anaconda (Python 2 is not supported). Make sure numpy is installed. Hope it helps.

I have checked the CMakeCache.text in the build directory:
![1631932187(1)](https://user-images.githubusercontent.com/59586135/133869390-4da281f0-5353-4494-bcca-2038c1811881.png)

It seems that only **the Numpy directory is not found on line 1103**, so I manually specified the numpy directory with other information as you suggested:

$> cmake -D CMAKE_BUILD_TYPE=RELEASE -DBUILD_TIFF=ON \
$> -DPYTHON_LIBRARY=/opt/anaconda3/lib/libpython3.7m.so \
$> -DPYTHON_INCLUDE_DIR=/opt/anaconda3/include/python3.7m \
$> -DPYTHON_EXECUTABLE=/opt/anaconda3/bin/python \
**$> -DPython_NumPy_INCLUDE_DIR=/opt/anaconda3/lib/python3.7/site-packages/numpy/core/include/numpy \
$> -DADIOS2_USE_Python=ON** ../

At least, cmake runs to the very end of the compilation. But, I'm not sure if adios2 is "correctly" compiled in this way. 

Basically, the original error appeared when I turn on "use python" by the option **-DADIOS2_USE_Python=ON**.

**Comment by williamfgc** [comment]
> At least, cmake runs to the very end of the compilation. But, I'm not sure if adios2 is "correctly" compiled in this way.

As long as CMake finds the intended dependencies (the config step will tell if Python bindings are enabled at end) and you can build and run, it should be fine. Keep in mind this is a non-standard configuration mixing package managers, so there is no correct or wrong. Hope it helps.

**Comment by alanwillwong** [comment]
> > At least, cmake runs to the very end of the compilation. But, I'm not sure if adios2 is "correctly" compiled in this way.
> 
> As long as CMake finds the intended dependencies (the config step will tell if Python bindings are enabled at the end) and you can build and run, it should be fine. Keep in mind this is a non-standard configuration mixing package managers, so there is no correct or wrong. Hope it helps.

I see. Thank you for your help.
